pandas/core/frame.py

```diff
@@ -1399,11 +1399,44 @@ def to_dict(self, orient="dict", into=dict):
             )
         # GH16122
         into_c = com.standardize_mapping(into)
-        if orient.lower().startswith("d"):
+        orient = orient.lower()
+        # GH32515
+        if orient.startswith(("d", "l", "s", "r", "i")) and orient not in {
+            "dict",
+            "list",
+            "series",
+            "split",
+            "records",
+            "index",
+        }:
+            warnings.warn(
+                "Using short name for 'orient' is deprecated. Only the "
+                "options: ('dict', list, 'series', 'split', 'records', 'index') "
+                "will be used in a future version. Use one of the above "
+                "to silence this warning.",
+                FutureWarning,
+            )
+
+        if orient.startswith("d"):
```

Comment:
you can make a dict instead of an if/else for this

Reply:
I'm not sure. If I create a dict such as :
```
orient_dict = {"d": "dict", 
 "l": "list", 
 "sp": "split", 
 "s": "series", 
 "r": "records", 
 "i": "index"}
```

We can not run our dictionary directly as `dict[orient]` because the options such as `dict` or even `racoon` (see the issue) would break. 

If we decide only to take the first letter (i.e. `orient_dict[orient[0]]`), this would not work for `split`, as we would still have to set an if statement  saying `if orient.startswith("sp")`. 

Unless there is something I am missing

Reply:
@jreback what's your opinion on the above?
I could change the code to something like:
´´´
orient_dict = {"d": "dict", 
 "l": "list", 
 "sp": "split", 
 "s": "series", 
 "r": "records", 
 "i": "index"}

if orient[0:2]=="sp":
    orient = "split"
else:
    orient = orient_dict[orient[0]]
´´´
or something similar. But in my opinion I'd rather have the current if statements.